app/uk/gov/hmrc/hec/models/fileFormat/FileBody.scala

```diff
@@ -27,3 +27,50 @@ final case class EnumFileBody(recordType: String = "01", recordId: String, recor
   override def toRowString: String = this.productIterator.mkString("|")
 
 }
+
+final case class HECTaxCheckFileBody(
+  recordType: String = "01",
+  ggCredID: Option[String] = None,
+  nino: Option[String] = None,
+  firstName: Option[String] = None,
+  lastName: Option[String] = None,
+  dob: Option[String] = None,
+  SAUTR: Option[Int] = None,
+  CTUTR: Option[String] = None,
+  crn: Option[String] = None,
+  companyName: Option[String] = None,
+  licenceType: String,
+  licenceValidityPeriod: String,
+  licenceTimeTrading: String,
+  entityType: Char,
+  notChargeable: Option[Char] = None,
+  PAYE: Option[Char] = None,
+  SA: Option[Char] = None,
+  CT: Option[Char] = None,
+  incomeTaxYear: Option[Int] = None,
+  hasAccountingPeriod: Option[Char] = None,
+  accountingPeriodStartDate: Option[String] = None,
+  accountingPeriodEndDate: Option[String] = None,
+  recentlyStartedTrading: Option[Char] = None,
+  returnReceived: Option[Char] = None,
+  noticeToFile: Option[Char] = None,
+  taxComplianceDeclaration: Option[Char] = None,
+  correctiveAction: Option[String] = None,
+  customerDeclaration: Char,
+  taxCheckStartDateTime: String,
+  taxCheckCompleteDateTime: String,
+  taxCheckCode: String,
+  taxCheckExpiryDate: String,
+  onlineApplication: Char
+) extends FileBody {
+
+  @SuppressWarnings(Array("org.wartremover.warts.Any"))
+  override def toRowString: String = this.productIterator
+    .map {
+      case Some(value) => value
+      case None        => "null"
```

Comment:
null in the file means empty string - do it'd just be `""` here

Reply:
oh ok , i misunderstood it 🤦‍♀️